COOPR-516 Move a cluster to the 'terminated' state instead of

albertshau opened the pull request:
'incomplete' if all create tasks failed in this manner.

albertshau committed:
COOPR-516 Move a cluster to the 'terminated' state instead of
'incomplete' if all create tasks failed in this manner.

poornachandra commented on coopr-server/src/main/java/co/cask/coopr/scheduler/JobScheduler.java:
```diff
@@ -178,8 +184,15 @@ public void run() {
-            taskService.failJobAndSetClusterStatus(job, cluster);
+            // special case: if all tasks were create tasks and all of them failed before they created anything,
+            // set the cluster state to 'terminated' instead of letting it go to 'incomplete'.
+            if (createFailedTasks == currentStage.size()) {
```
It would be safer to check if none of the creates succeeded here to make sure there are no edge cases.

poornachandra commented:
Done with review, please address the comment.

albertshau committed:
COOPR-516 changing the logic a bit to be more robust. As an added
bonus, jobs that are aborted before any task could be taken are
also moved to terminated instead of incomplete.

poornachandra commented:
LGTM :+1:
